rio/srcrobot/robot_container.py

```diff
@@ -164,19 +164,19 @@ def configureButtonBindings(self):
         #Que Buttons
         self.queSubFront.onTrue(InstantCommand(lambda: self.m_robotState.m_gameState.setNextShot(
             Constants.NextShot.SPEAKER_CENTER
-        )))
+        )).andThen(self.s_Arm.seekArmZero()))
```

Comment:
Why add arm to zero to these? Adding arm to zero after shooting makes sense but why after just setting the next shot?

Reply:
Raza really wanted to be able to re-execute when the shot is changed. For example if we queue the amp shit but decide we want to pivot to the speaker for some reason, we wouldn't need to shoot to cancel the execution. The only way I could find to do that was to schedule a command that cancels one of the commands in the execute, which would cancel all of the commands on the execute. I don't think it would make much of a difference in practice though because amp->speaker or speaker->amp both need the arm to be down at some point.

Reply:
He did say if this makes it awkward we can omit it

Reply:
The execute should just react properly to the selected shot changing. No reset of any kind should be needed. This comment prompted me to take another hard look at the execute command and it has some serious issues. I will comment in there next.

Reply:
For the execute command raza wanted to have it such that, while executing, the driver can still control the turn, but when the turn stick is let go the robot snaps back to the required angle. He also wanted the execute to hold the position of everything until it is explicitly by a shoot, queue, or autohome

Reply:
I see how TurnInPlace was changed for this consideration and that is good. However the suppliers defined above for translation, strafe and rotation are not having a deadband applied. Combine this with comparing to 0.0 for rotation input in TurnInPlace and it may never snap to the proper angle.

Reply:
Also, this would be an okay way to try first. However, ExecuteCommand still needs a rewrite as I described in there, but use option 1 rather than option 2.